### PR TITLE
handle e.g. empty dict body data properly

### PR DESCRIPTION
**Pull request by stdweird**

### CODE REVIEW

File: lib/vsc/utils/rest.py [hunk 1]
```diff
@@ -172,18 +172,20 @@ def request(self, method, url, body, headers, content_type=None):
...
-        fancylogger.getLogger().debug('cli request: %s, %s, %s, %s', method, url, body_censored, headers_censored)
+        body_censored = body
+        if body is not None:
+            if isinstance(body, str):
+                # assume serialized bodies are already clear of secrets
+                logging.debug("Request with pre-serialized body, will not censor secrets")
+                body_censored = body
+            else:
+                # censor contents of body to avoid leaking passwords
+                secret_items = ['password']
```
Comment (wpoely86) [inline lib/vsc/utils/rest.py:183]: This is already defined on L172?

Comment (wpoely86) [reply lib/vsc/utils/rest.py:184]: Nevermind, misread

File: lib/vsc/utils/rest.py [hunk 2]
```diff
@@ -172,18 +173,20 @@ def request(self, method, url, body, headers, content_type=None):
...
-            fancylogger.getLogger().debug("Request with pre-serialized body, will not censor secrets")
-            body_censored = body
-
-        fancylogger.getLogger().debug('cli request: %s, %s, %s, %s', method, url, body_censored, headers_censored)
+        body_censored = body
+        if body is not None:
+            if isinstance(body, str):
+                # assume serialized bodies are already clear of secrets
+                logging.debug("Request with pre-serialized body, will not censor secrets")
+                body_censored = body
```
Comment (wpoely86) [inline lib/vsc/utils/rest.py:181]: This is whole `if` still needed if you already set it on L176?

File: lib/vsc/utils/rest.py (hunk 1 above)
Comment (stdweird) [reply lib/vsc/utils/rest.py:184]: it reuses a variable name, but a bit odd to read indeed

File: lib/vsc/utils/rest.py (hunk 2 above)
Comment (stdweird) [reply lib/vsc/utils/rest.py:181]: what do you mean?

Comment (wpoely86) [reply lib/vsc/utils/rest.py:181]: before the `if` you already do `body_censored = body`, so except for the debug line, this branch does nothing?

Comment (stdweird) [reply lib/vsc/utils/rest.py:181]: well, the log in relevant, but i indeed forgot to remove this line

**Review by wpoely86 — APPROVED** [review]
Good for me but I have no merge rights.

**Comment by wdpypere** [comment]
@stdweird I reintroduced the old pylint hack to vsc-install, this should fix the test issues you were running into.

**Pull request merged**